packages/typespec-test/test/batch_modular/generated/typespec-ts/src/rest/batchClient.ts

```diff
@@ -18,7 +18,7 @@ export default function createClient(
   options: ClientOptions = {},
 ): BatchContext {
   const baseUrl = options.baseUrl ?? `${endpoint}`;
-  options.apiVersion = options.apiVersion ?? "2023-05-01.17.0";
```

Comment:
this is expected, not all operations in batch spec contains an api-version parameter. we should not set the default value for it.